### PR TITLE
docs: add sorbet documentation

### PR DESCRIPTION
**Pull request by vidusheeamoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
Documentation for Sorbet.

### CODE REVIEW

**Comment by vidusheeamoli** [comment]
Going to add another section which deals with how to resolve type errors reported by Sorbet ASAP.
cc - @issyl0 @mistydemeo

File: docs/Typechecking.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The [`sorbet/`](https://github.com/Homebrew/brew/tree/master/Library/Homebrew/sorbet)
+directory in `brew/Library/Homebrew` consists of:
+
+- The `rbi/` directory. It contains all Ruby Interface files, which help Sorbet to
+learn about constants, ancestors, and methods defined in ways it doesn’t understand
+natively. RBI files for all gems are auto-generated using
+[Tapioca](https://github.com/Shopify/tapioca#tapioca). We can also create a RBI
+file to help Sorbet understand dynamic definitions.
+For eg: Sorbet assumes that `Kernel` is not necessarily included in our modules
```
Comment (MikeMcQuaid) [inline docs/Typechecking.md:18]:
```suggestion
For example: Sorbet assumes that `Kernel` is not necessarily included in our modules
```

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good to me so far! Can you also link this in the README under "Contributors"? Thanks!

**Comment by MikeMcQuaid** [comment]
Let's merge for now and can always address additional comments in a follow-up PR. Nice work again @vidusheeamoli!

**Pull request merged**